diff --git a/src/llvm/cmake/config-ix.cmake b/src/llvm/cmake/config-ix.cmake
index 818fafb..023612f 100644
--- a/src/llvm/cmake/config-ix.cmake
+++ b/src/llvm/cmake/config-ix.cmake
@@ -449,6 +449,8 @@ elseif (LLVM_NATIVE_ARCH MATCHES "riscv32")
   set(LLVM_NATIVE_ARCH RISCV)
 elseif (LLVM_NATIVE_ARCH MATCHES "riscv64")
   set(LLVM_NATIVE_ARCH RISCV)
+elseif (LLVM_NATIVE_ARCH MATCHES "loongarch")
+  set(LLVM_NATIVE_ARCH LoongArch)
 else ()
   message(FATAL_ERROR "Unknown architecture ${LLVM_NATIVE_ARCH}")
 endif ()
